includes/MyWikisWikisPager.php

```diff
@@ -0,0 +1,175 @@
+	private $state;
+
+	/** @var WikiDiscover */
+	private $wikiDiscover;
+
+	/**
+	 * @param SpecialPage $page
+	 * @param string $language
+	 * @param string $category
+	 * @param string $state
+	 */
+	public function __construct( $page, $language, $category, $state ) {
+		$this->mDb = self::getCreateWikiDatabase();
+
+		$this->language = $language;
+		$this->category = $category;
+
+		$this->state = $state;
+
+		$this->wikiDiscover = new WikiDiscover();
+
+		parent::__construct( $page->getContext(), $page->getLinkRenderer() );
+	}
+
+	/**
+	 * @return IReadableDatabase
+	 */
+	public static function getCreateWikiDatabase() {
+		$connectionProvider = MediaWikiServices::getInstance()->getConnectionProvider();
+		return $connectionProvider->getReplicaDatabase( 'virtual-createwiki', 'cw_wikis' );
+	}
+
+	/** @inheritDoc */
+	protected function getFieldNames() {
+		$config = MediaWikiServices::getInstance()->getMainConfig();
+
+		static $headers = null;
+
+		$headers = [
+			'cw_dbname' => 'wikidiscover-table-wiki',
+			'cw_language' => 'wikidiscover-table-language',
+		];
+
+		$headers += [
+			'cw_category' => 'wikidiscover-table-category',
+			'cw_timestamp' => 'wikidiscover-table-established',
+		];
+
+		if ( ExtensionRegistry::getInstance()->isLoaded( 'ManageWiki' ) && $this->getConfig()->get( 'WikiDiscoverUseDescriptions' ) ) {
+			$headers['wiki_description'] = 'wikidiscover-table-description';
+		}
+
+		foreach ( $headers as &$msg ) {
+			$msg = $this->msg( $msg )->text();
+		}
+
+		return $headers;
+	}
+
+	/** @inheritDoc */
+	public function formatRow( $row ) {
+		if ( ExtensionRegistry::getInstance()->isLoaded( 'ManageWiki' ) ) {
+			$manageWikiSettings = new ManageWikiSettings( $row->wiki_dbname );
+			if ( $manageWikiSettings->list( 'wgWikiDiscoverExclude' ) ) {
+				return '';
+			}
+		}
+
+		return parent::formatRow( $row );
+	}
+
+	/** @inheritDoc */
+	public function formatValue( $name, $value ) {
+		$row = $this->mCurrentRow;
+
+		$wikiDiscover = $this->wikiDiscover;
+
+		$wiki = $row->cw_dbname;
+
+		switch ( $name ) {
+			case 'cw_dbname':
+				$url = $wikiDiscover->getUrl( $wiki );
+				$name = $wikiDiscover->getSitename( $wiki );
+				$formatted = "<a href=\"{$url}\">{$name}</a>";
+				break;
+			case 'cw_language':
+				$formatted = $wikiDiscover->getLanguage( $wiki );
+				break;
+			case 'cw_private':
+				if ( $row->cw_category === 1 ) {
+					$formatted = 'Private';
+				} else {
+					$formatted = 'Public';
+				}
+				break;
+			case 'cw_category':
+				$wikiCategories = array_flip( $this->getConfig()->get( 'CreateWikiCategories' ) );
+				$formatted = $wikiCategories[$row->cw_category] ?? 'Uncategorised';
+				break;
+			case 'cw_timestamp':
+				$lang = RequestContext::getMain()->getLanguage();
+
+				$formatted = htmlspecialchars( $lang->date( wfTimestamp( TS_MW, strtotime( $row->cw_timestamp ) ) ) );
+				break;
+			default:
+				$formatted = "Unable to format $name";
+				break;
+		}
+
+		return $formatted;
+	}
```

Comment:
Escape _everything_ as a safety precaution per [MediaWiki wiki recommendation](https://www.mediawiki.org/wiki/Cross-site_scripting#Stopping_Cross-site_scripting), such as in https://github.com/miraheze/WikiDiscover/blob/2b1ae41826e0100e774367c64d67f540e5a30ea1/includes/WikiDiscoverWikisPager.php#L106-L175